modules/vault-cluster/main.tf

```diff
@@ -294,6 +306,26 @@ data "aws_iam_policy_document" "vault_s3" {
   }
 }
 
+data "aws_iam_policy_document" "vault_dynamo" {
+  count = var.enable_dynamo_backend ? 1 : 0
+
+  statement {
+    effect    = "Allow"
+    actions   = ["dynamodb:*"]
```

Comment:
Please limit to the specific permissions necessary: https://www.vaultproject.io/docs/configuration/storage/dynamodb/#required-aws-permissions